shaders/palette.frag

```diff
@@ -42,6 +43,33 @@ vec4 handle(float index, float remap) {
 }
 
 void main() {
+    // Don't render if we're decimating due to opacity
+    float decimate_limit = opacity / 255.0;
+    float decimate_value = noise(gl_FragCoord.xy);
+    if (decimate_value > decimate_limit) {
+        discard;
+    }
+
+    if (SPRITE_SHADOWMASK) {
+        // make four samples to generate coverage
+        int coverage = 0;
+        for(int y = 0; y < 4; y++) {
+            float offset = float(y) / ATLAS_H;
+            vec4 texel = texture(atlas, tex_coord + vec2(0, offset));
+            int index = int(texel.r * 255.0);
+            coverage += int(index != transparency_index);
+        }
+
+        // don't draw transparent pixels
+        if(coverage == 0) {
+            discard;
+        }
+
+        color = handle(float(coverage) / 256.0);
+        return;
+    }
```

Comment:
Hum. On one hand, I like it that this is the graphics backends problem. On the other hand, I wonder what would be the best way if we implement the software renderer at some point.

Reply:
In the other branch, I lazily compute a squashed "shadow" surface which is no more expensive to draw than any other surface.

Reply:
Let's just merge this one. We can always tinker with it if we need to.